go/README.md

```diff
@@ -4,27 +4,70 @@ This builder runs the `go` tool (`go build`, `go test`, etc.)
 after placing source in `/workspace` into the `GOPATH` before
 running the tool.
 
-This functionality is not necessary if you're building using
+# Using `Golang` and [Go Modules](https://github.com/golang/go/wiki/Modules)
+
+This Builder (`gcr.io/cloud-builders/go`) is not necessary if you're building using
 [Go modules](https://github.com/golang/go/wiki/Modules), available
-in Go 1.11+, and you can **build with the standard
-[`golang`](https://hub.docker.com/_/golang) image on Dockerhub
-instead:**
+in Go 1.11+. You can **build** with the `golang` image (not a Builder) from [Dockerhub](hub.docker.com/library/golang) and Google's Container Registry [mirror](mirror.gcr.io/library/golang):
 
 ```
 steps:
 # If you already have a go.mod file, you can skip this step.
-- name: golang
+- name: mirror.gcr.io/library/golang
   args: ['go', 'mod', 'init', 'github.com/your/import/path']
 
 # Build the module.
-- name: golang
+- name: mirror.gcr.io/library/golang
   env: ['GO111MODULE=on']
   args: ['go', 'build', './...']
 ```
 
 See [`examples/module`](https://github.com/GoogleCloudPlatform/cloud-builders/tree/master/go/examples/module)
 for a working example.
 
+## Note #1 `/workspace` and `/go`
+The `Golang` image defaults to a working directory of `/go` whereas Cloud Build mounts your sources under `/workspace`. This reflects the recommend best practice when using Modules of placing your sources *outside* of `GOPATH`. When you `go build ./...` you will run this in the working directory of `/workspace` *but* the packages will be pulled into `/go/pkg`.
+
+Because `/go` is outside of `/workspace`, the `/go` directory is not persisted across Cloud Build steps. See next note.
+
+## Note #2 Sharing packages across steps
+
+One advantage with Go Modules is that packages are now semantically versioned and immutable; one a package has been pulled once, it should not need to be pulled again. Because the `golang` image uses `/go` as its working directory and this is outside of Cloud Build's `/workspace` directory, `/go` is recreated in each `Golang` step. To avoid this and share packages across steps, you may use Cloud Build `volumes`. An example to prove the point:
+
+```YAML
+- name: mirror.gcr.io/golang
+  env:
+  - GO111MODULE=on
+  args: ['go','get','-u','github.com/golang/glog']
+  volumes:
```

Comment:
You can now specify build-wide volumes and envs using `.options.env` and `.options.volumes`, to avoid having to repeat yourself.

Reply:
IDNKT. Will reflect.

Reply:
See https://github.com/GoogleCloudPlatform/cloud-builders/pull/517/commits/f7fbd8a23bdb6a3be7e5911df90506958a525852